### DISCUSSION TITLE
Breaking Change Management - Is it only for Software/Tool versions?

### DISCUSSION THREAD

**Discussion by wsmelton** [actions/virtual-environments#8631] [General questions] [answered]
I'm only asking because the last breaking change with the VHD to Managed Identity took my team a few weeks to rework. Is there a process of announcing breaking changes to the structure of this repository?

I see two PRs that could be merged at any point that would be breaking for those customers that consume this repository to be self-hosted images for Azure DevOps (or GitHub too).

- [Windows] Implement new directories hierarchy #8616
- [Ubuntu] Implement new directories hierarchy #8627

The above change the path to the post-generation scripts for both OS types. Which those customers that consume this repository from latest of main and push post-generation scripts through this to their images will break. We expect the path to be something like `images/$(OSName)/post-generation` but these PRs will change that to `images/$(OSName)/assets/post-gen`.

**Comment by vpolikarpov-akvelon** [comment, marked as answer]
Hey @wsmelton. We've posted an announcement [here](https://github.com/actions/runner-images/issues/8706). Please take a look.
